tests/e2e/util.go

```diff
@@ -5106,3 +5106,112 @@ func waitForVolumeSnapshotContentToBeDeleted(client snapclient.Clientset, ctx co
 	})
 	return waitErr
 }
+
+// getK8sMasterNodeIPWhereControllerLeaderIsRunning fetches the master node IP
+// where controller is running
+func getK8sMasterNodeIPWhereContainerLeaderIsRunning(ctx context.Context,
+	client clientset.Interface, sshClientConfig *ssh.ClientConfig,
+	containerName string) (string, string, error) {
+	ignoreLabels := make(map[string]string)
+	var k8sMasterNodeIP string
+	csiControllerPod, grepCmdForFindingCurrentLeader := "", ""
+	csiPods, err := fpod.GetPodsInNamespace(client, csiSystemNamespace, ignoreLabels)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	k8sMasterIPs := getK8sMasterIPs(ctx, client)
+	k8sMasterIP := k8sMasterIPs[0]
+	for _, csiPod := range csiPods {
+		if strings.Contains(csiPod.Name, vSphereCSIControllerPodNamePrefix) {
+			// Fetching the name of CSI controller pod where container leader is running
```

Comment:
also add a note that it is not valid for csi `vsphere-csi-controller` container, since in case of `vsphere-csi-controller` all the replicas will behave as leaders